src/EditModel/SectionDialog.component.js

```diff
@@ -26,176 +29,263 @@ class SectionDialog extends React.Component {
 
         this.state = {
             categoryCombo: false,
-            disableDataElementAutoGroup: false
+            disableDataElementAutoGroup: false,
+            displayOptions: {
+                pivotMode: null,
+                pivotedCategory: null,
+            },
+            categories: [],
         };
         dataElementStore.setState([]);
         assignedDataElementStore.setState([]);
         indicatorStore.setState([]);
         assignedIndicatorStore.setState([]);
 
-        this.getTranslation = context.d2.i18n.getTranslation.bind(context.d2.i18n);
+        this.getTranslation = context.d2.i18n.getTranslation.bind(
+            context.d2.i18n
+        );
     }
 
     componentDidMount() {
         this.subscriptions = [];
-        this.subscriptions.push(assignedDataElementStore.subscribe(() => {
-            this.forceUpdate();
-        }));
+        this.subscriptions.push(
+            assignedDataElementStore.subscribe(() => {
+                this.forceUpdate();
+            })
+        );
+
+        const d2 = this.context.d2;
+        const categoryCombos = this.props.categoryCombos
+            .map(coc => coc.value)
+            .join(',');
+
+        return d2.models.categoryCombos
```

Comment:
You should be able to get this data directly from `categories`-endpoint instead of having to gather the categories from the catcombos yourself.

Something like this: `https://debug.dhis2.org/dev/api/categories.json?fields=id,displayName,categoryCombos[id]&filter=categoryCombos.id:in:[DJXmyhnquyI,bjDvmb4bfuf]`


```js

return d2.models.categories.list({ filter: `categoryCombos.id:in:[${categoryCombos}]`, fields: ['id,displayName] })

```

```

Reply:
I also think we should filter out the default category, right? It doesn't make much sense to select that, right? The default category can only be by itself, and thus you should just pivot the form instead of moving it.

Reply:
I updated the two comments (about fetching and being defensive) - thanks for these!

Regarding the default category, it's a bit fiddly because sections can have data elements with different category combos, so it's not necessarily clear whether it's the only option or not for a specific section. I need to figure that usecase better, and I will have a followup PR if that's ok - for now, they should be able to "preview" the change and see if something is wrong with the configuration until we tighten the option per section better.